From: Michael Olbrich <m.olbrich@pengutronix.de>
Subject: HACK: revert 24355313dc9686a14a618fcd9bbdedc29c57b651 (readd bsg.h)

Readd bsg.h which is not part of the kernel headers in OSELAS.Toolchain-1.99.3

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

---
 Makefile.am                  |    3 +
 extras/scsi_id/bsg.h         |   84 +++++++++++++++++++++++++++++++++++++++++++
 extras/scsi_id/scsi_serial.c |    3 +
 3 files changed, 88 insertions(+), 2 deletions(-)

Index: udev-162/Makefile.am
===================================================================
--- udev-162.orig/Makefile.am
+++ udev-162/Makefile.am
@@ -287,7 +287,8 @@ extras_scsi_id_scsi_id_SOURCES =\
 	extras/scsi_id/scsi_id.c \
 	extras/scsi_id/scsi_serial.c \
 	extras/scsi_id/scsi.h \
-	extras/scsi_id/scsi_id.h
+	extras/scsi_id/scsi_id.h \
+	extras/scsi_id/bsg.h
 extras_scsi_id_scsi_id_LDADD = libudev/libudev-private.la
 libexec_PROGRAMS += extras/scsi_id/scsi_id
 dist_man_MANS += extras/scsi_id/scsi_id.8
Index: udev-162/extras/scsi_id/bsg.h
===================================================================
--- /dev/null
+++ udev-162/extras/scsi_id/bsg.h
@@ -0,0 +1,84 @@
+#ifndef BSG_H
+#define BSG_H
+
+#define BSG_PROTOCOL_SCSI		0
+
+#define BSG_SUB_PROTOCOL_SCSI_CMD	0
+#define BSG_SUB_PROTOCOL_SCSI_TMF	1
+#define BSG_SUB_PROTOCOL_SCSI_TRANSPORT	2
+
+struct sg_io_v4 {
+	__s32 guard;		/* [i] 'Q' to differentiate from v3 */
+	__u32 protocol;		/* [i] 0 -> SCSI , .... */
+	__u32 subprotocol;	/* [i] 0 -> SCSI command, 1 -> SCSI task
+				   management function, .... */
+
+	__u32 request_len;	/* [i] in bytes */
+	__u64 request;		/* [i], [*i] {SCSI: cdb} */
+	__u64 request_tag;	/* [i] {SCSI: task tag (only if flagged)} */
+	__u32 request_attr;	/* [i] {SCSI: task attribute} */
+	__u32 request_priority;	/* [i] {SCSI: task priority} */
+	__u32 request_extra;	/* [i] {spare, for padding} */
+	__u32 max_response_len;	/* [i] in bytes */
+	__u64 response;		/* [i], [*o] {SCSI: (auto)sense data} */
+
+        /* "dout_": data out (to device); "din_": data in (from device) */
+	__u32 dout_iovec_count;	/* [i] 0 -> "flat" dout transfer else
+				   dout_xfer points to array of iovec */
+	__u32 dout_xfer_len;	/* [i] bytes to be transferred to device */
+	__u32 din_iovec_count;	/* [i] 0 -> "flat" din transfer */
+	__u32 din_xfer_len;	/* [i] bytes to be transferred from device */
+	__u64 dout_xferp;	/* [i], [*i] */
+	__u64 din_xferp;	/* [i], [*o] */
+
+	__u32 timeout;		/* [i] units: millisecond */
+	__u32 flags;		/* [i] bit mask */
+	__u64 usr_ptr;		/* [i->o] unused internally */
+	__u32 spare_in;		/* [i] */
+
+	__u32 driver_status;	/* [o] 0 -> ok */
+	__u32 transport_status;	/* [o] 0 -> ok */
+	__u32 device_status;	/* [o] {SCSI: command completion status} */
+	__u32 retry_delay;	/* [o] {SCSI: status auxiliary information} */
+	__u32 info;		/* [o] additional information */
+	__u32 duration;		/* [o] time to complete, in milliseconds */
+	__u32 response_len;	/* [o] bytes of response actually written */
+	__s32 din_resid;	/* [o] din_xfer_len - actual_din_xfer_len */
+	__s32 dout_resid;	/* [o] dout_xfer_len - actual_dout_xfer_len */
+	__u64 generated_tag;	/* [o] {SCSI: transport generated task tag} */
+	__u32 spare_out;	/* [o] */
+
+	__u32 padding;
+};
+
+#ifdef __KERNEL__
+
+#if defined(CONFIG_BLK_DEV_BSG)
+struct bsg_class_device {
+	struct device *class_dev;
+	struct device *parent;
+	int minor;
+	struct request_queue *queue;
+	struct kref ref;
+	void (*release)(struct device *);
+};
+
+extern int bsg_register_queue(struct request_queue *q,
+			      struct device *parent, const char *name,
+			      void (*release)(struct device *));
+extern void bsg_unregister_queue(struct request_queue *);
+#else
+static inline int bsg_register_queue(struct request_queue *q,
+				     struct device *parent, const char *name,
+				     void (*release)(struct device *))
+{
+	return 0;
+}
+static inline void bsg_unregister_queue(struct request_queue *q)
+{
+}
+#endif
+
+#endif /* __KERNEL__ */
+
+#endif
Index: udev-162/extras/scsi_id/scsi_serial.c
===================================================================
--- udev-162.orig/extras/scsi_id/scsi_serial.c
+++ udev-162/extras/scsi_id/scsi_serial.c
@@ -32,7 +32,8 @@
 #include <scsi/scsi.h>
 #include <scsi/sg.h>
 #include <linux/types.h>
-#include <linux/bsg.h>
+/* #include <linux/bsg.h> */
+#include "bsg.h"
 
 #include "libudev.h"
 #include "libudev-private.h"
